UI/SitRepPanel.cpp

```diff
@@ -202,12 +206,16 @@ namespace {
                 DoLayout();
             }
             if (m_link_text) {
-                // Use the height of the text as our height.
+                // establish sitrep panel's size; use icon or text size for panel height, whichever is larger
                 // DoLayout reflowed the text.
-                GG::Pt text_size = m_link_text->TextLowerRight() - m_link_text->TextUpperLeft();
-                text_size.y += ITEM_VERTICAL_PADDING*2; // Text centers, so this puts padding on both above and below
-                text_size.x = lr.x - ul.x; // Ignore the width of the text, use whatever was requested.
-                GG::Control::SizeMove(ul, ul + text_size );
+                GG::Pt panel_text = m_link_text->TextLowerRight() - m_link_text->TextUpperLeft();
+                int text_size = Value(panel_text.y);
```

Comment:
"panel_text" seems like a wrong name for something that is more of a text size, and you could simply combine these lines: 
```
int text_size = Value((m_link_text->TextLowerRight() - m_link_text->TextUpperLeft()).y);
```

Also, just to mention an alternative approach using different info that you might more specifically want to use sometime, you can also get a working result with 
```
int text_size = Value(m_link_text->GetFont()->Lineskip()) * m_link_text->GetLineData().size();
```